### DISCUSSION TITLE
WinUI @Build2024!

### DISCUSSION THREAD

**Discussion by duncanmacmichael** [microsoft/microsoft-ui-xaml#9649] [General]
//Build 2024 is underway, and WinUI has a big presence this year! We've got several breakout sessions lined up, and a live demo for those attending in-person. Here are links to the great content we've got lined up for you this year - we'll update this post as more becomes available, so make sure to check this space!

### //Build 2024 sessions and announcements

- See all the new //Build announcements on the //Build Blog here, including the latest details about WinUI: [Unlock a new era of innovation with Windows Copilot Runtime and Copilot+ PCs](https://blogs.windows.com/windowsdeveloper/2024/05/21/unlock-a-new-era-of-innovation-with-windows-copilot-runtime-and-copilot-pcs/)
- Learn how to choose the right tech for your Win32 app development on Windows, with Chigusa and Jeffrey: [Navigating Win32 App Development with WinUI and WPF](https://build.microsoft.com/en-US/sessions/11626139-a9d0-4f8c-b664-3f3436cea50a)
  - This session is scheduled for Wednesday, May 22nd at 4:45 PM PDT (23:45 UTC)
- See how you can elevate your app-building skills with Niels, Ranjesh, and Pankaj: [How to create superior experiences with WinUI and WPF](https://build.microsoft.com/en-US/sessions/5c7afbc3-1e4c-41bb-b883-b66f78b46be7)
  - This session is scheduled for Thursday, May 23rd at 3:00 PM PDT (22:00 UTC)
- If you're in-person in Seattle at //Build, check out the live demo on how to make a great, theme-responsive titlebar with WinUI! [Elevating app craftsmanship with WinUI 3](https://build.microsoft.com/en-US/sessions/c326047b-7cf1-4233-b8a3-8ada6e5b891f)
  - This demo is scheduled for Tuesday, May 21st at 1:00 PM PDT
- Discover how to empower your WinUI apps with AI with Nikola and Tetyana: [Use AI for "real things" in your Windows Apps](https://build.microsoft.com/en-US/sessions/78618df5-6b18-401c-8df0-550263168f39)
  - This session is scheduled for Tuesday, May 21st at 4:45 PM PDT (23:45 UTC)
- Visual Studio 17.10 was announced at //Build, and with it comes a new WinUI workload, new WinUI Unit Test templates, and an updated WinUI template selection experience for new projects. Check out the [release notes](https://learn.microsoft.com/en-us/visualstudio/releases/2022/release-notes#new-winui-workload-and-template-improvements) to learn more and download Visual Studio to check out the new WinUI changes!

### New and updated links

- We've got a brand new Windows Dev Center page for WinUI, WPF, and more! This page helps new developers see what WinUI can do and get started with their first WinUI app faster than ever before. Check it out here: https://aka.ms/windev
- We've revamped our Learn documentation with a new Getting Started topic, a refreshed topic for learning about the different framework options, and a reorganized table of contents. Check them out here:
  - [Get started with WinUI](https://learn.microsoft.com/en-us/windows/apps/get-started/start-here?tabs=vs-2022-17-10)
  - [Overview of framework options](https://learn.microsoft.com/en-us/windows/apps/get-started/?tabs=cpp-win32%2Cnet-maui)
- Don't forget to check out our gallery app, too! https://aka.ms/WinUIGallery

We've also updated the [ReadMe](https://github.com/microsoft/microsoft-ui-xaml) in this repo to make it cleaner and clearer, so be sure to take a look!

**Comment by kmgallahan** [comment]
Hey @duncanmacmichael, do you know if this new VS feature will work with C# WinUI 3 projects when .NET 9 is supported? 

"The debugger now breaks on async method exceptions caught by framework code."

https://learn.microsoft.com/en-us/visualstudio/releases/2022/release-notes-preview#asyncexceptionbreak

https://developercommunity.visualstudio.com/t/Provide-tooling-in-Visual-Studio-to-brea/1346245

**Reply by duncanmacmichael** [reply]
Hi @kmgallahan, great question! We think this question is really more suited for the Visual Studio or .NET 9 teams, but we do anticipate that this feature will work the same in C# WinUI 3 projects as it does in other C# projects. Thanks!

**Comment by LoriBru** [comment]
It's funny (not so much, actually) to see messages like this being given even at Microsoft Build, directly!
[How to Quickly Build a .NET Desktop Dashboard](https://build.microsoft.com/en-US/sessions/2e7fbd78-8829-4bea-b097-e9841a154081)
![Screenshot 2024-05-28 115059](https://github.com/microsoft/microsoft-ui-xaml/assets/22811725/0f7b787f-8349-4bca-adf4-c531de002569)

**Reply by mdtauk** [reply]
Mescius probably has a bunch of WPF controls and tooling they are pushing, so this message makes sense to them.

Build was focusing on showing how their 3 frameworks were improving support for Windows 11 styling, but WinUI was pushed as the latest thing that Windows is using for its new UI.

**Reply by pjmlp** [reply]
Or how C++ keeps being mentioned as a plus, without acknowledgong the reality of the sore state of its tooling versus what Qt and C++ Builder are capable of, in terms of developer experience.

**Comment by brabebhin** [comment]
Sorry but AI sessions don't count.

**Reply by NicholasChrzan** [reply]
True but at least WinUI was mentioned alongside with WPF and from the sessions wasn't completely ignored. The reality is that platform specific native development is no longer has the focus it once enjoyed and everyone is now chasing the single code base platform agnostic pipe dream.
